src/workers/parser/mapBindings.js

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+import { parseScript } from "./utils/ast";
+import { buildScopeList } from "./getScopes";
+import generate from "@babel/generator";
+import * as t from "@babel/types";
+import remove from "lodash/remove";
+
+import type { ColumnPosition } from "../../types";
+
+// translates new bindings `var a = 3` into `self.a = 3`
+// and existing bindings `var a = 3` into `a = 3` for re-assignments
+function globalizeDeclaration(node, bindings) {
+  return node.declarations.map(node => {
+    const identifier = bindings.includes(node.id.name)
+      ? node.id
+      : t.memberExpression(t.identifier("self"), node.id);
+
+    return t.expressionStatement(
+      t.assignmentExpression("=", identifier, node.init)
+    );
+  });
+}
+
+function globalizeAssignment(node, bindings) {
+  if (bindings.includes(node.left.name)) {
+    return node;
+  }
+
+  const identifier = t.memberExpression(t.identifier("self"), node.left);
+  return t.expressionStatement(
+    t.assignmentExpression("=", identifier, node.right)
```

Comment:
We should use `node.operator` instead of hard-coding `"="` since it this could also be `foo += 4` or something like that.